common/src/main/java/org/apache/uniffle/common/rpc/RpcUtils.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.common.rpc;
+
+import io.grpc.Status;
+import io.grpc.StatusException;
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.StreamObserver;
+import org.slf4j.Logger;
+
+import org.apache.uniffle.common.metrics.RPCMetrics;
+
+/**
+ * Utilities for handling server RPC calls.
+ *
+ * <p>There are two types of RPC calls: 1. RPCs that only throw IOException 2. Streaming RPCs For
+ * each of these, if failureOk is set, non-fatal errors will only be logged at the DEBUG level and
+ * failure metrics will not be recorded.
+ */
+public class RpcUtils {
```

Comment:
Perhaps `GrpcUtils` would be a more appropriate name, as it is solely utilized for gRPC operations.

Additionally, we plan to deprecate the use of gRPC for sending and receiving shuffle data. Therefore, it would be nice to implement similar metrics for Netty as well.